functions/mqhandlers.go

```diff
@@ -351,6 +351,15 @@ func updateHostConfig(host *models.Host) (resetInterface, restart bool) {
 	if host.MTU != 0 && hostCfg.MTU != host.MTU {
 		resetInterface = true
 	}
+	// do not update fields that should not be changed by server
+	host.OS = hostCfg.OS
+	host.FirewallInUse = hostCfg.FirewallInUse
+	host.DaemonInstalled = hostCfg.DaemonInstalled
+	host.ID = hostCfg.ID
+	host.Version = hostCfg.Version
+	host.MacAddress = hostCfg.MacAddress
+	host.PublicKey = hostCfg.PublicKey
```

Comment:
Does the public key refresh still work with this? If not tested, should be a part of the testing steps just to confirm.

Reply:
tested host key update;  confirmed feature still functions